### PR TITLE
Packaged the coverage report renderer for Hackage

### PR DESCRIPTION
**Pull request by SebastianKG**
Since it's not feasible to establish a WORKSPACE function for clients to build the coverage report renderer using a BUILD file inside of Rules_Haskell, the next best thing is to push it up to Hackage and let them pull it down themselves. This is the work necessary to make that possible.

### CODE REVIEW

File: tools/coverage-reports/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+If we have a target defined by the Rules_Haskell `haskell_test` rule, such as:
+
+```
+haskell_test(
+    name = "two-libs",
+    srcs = [
+        ...
+    ],
+    coverage_report_format = "html", # this must be set
```
Comment (Profpatsch) [inline tools/coverage-reports/README.md]: `# this must be set and can be one of …`

File: tools/coverage-reports/package.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+maintainer:          "m@tweag.io"
+copyright:           "2019 Tweag I/O Limited"
+
+extra-source-files:
+- LICENSE
+- package.yaml
+- README.md
+- stack.yaml
+
+synopsis:            Generate coverage report from Rules_Haskell bazel coverage output. See README at <https://github.com/tweag/rules_haskell/blob/master/tools/coverage-reports/README.md>
```
Comment (Profpatsch) [inline tools/coverage-reports/package.yaml]: You’re distributing the `README.md` with the source, so this link should not be needed.

File: tools/coverage-reports/package.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - cmdargs
+    - directory
+    - filepath
+    - hxt
+    - hxt-xpath
+    - listsafe
+    ghc-options:
+    - -rtsopts
+    - -threaded
+    - -with-rtsopts=-N
```
Comment (Profpatsch) [inline tools/coverage-reports/package.yaml]: Is this actually something which benefits from parallelizing?

Comment (SebastianKG) [reply tools/coverage-reports/package.yaml]: probably not. Removed

**Pull request merged**